pkg/controller/controller_broker.go

```diff
@@ -416,6 +414,12 @@ func (c *controller) updateServiceBrokerCondition(broker *v1alpha1.ServiceBroker
 		}
 	}
 
+	// Set status.ReconciledGeneration if updating ready condition to true
+
+	if conditionType == v1alpha1.ServiceBrokerConditionReady && status == v1alpha1.ConditionTrue {
```

Comment:
According to our face-to-face discussions, this should be changed to not depend on the ready condition being true.

Reply:
That's fine. It seems to me like this logic should go in `PrepareForUpdate(ctx, new, old)` in `pkg/registry/servicecatalog/broker/strategy.go`, though. That's where the previous checksum-setting logic lived.

Reply:
I did not want to put the ReconciledGeneration logic into the API server because I wanted to let the controller implement those semantics.